experimental/Schemes/IdealSheaves.jl

```diff
@@ -1210,3 +1210,105 @@ function saturation(I::IdealSheaf, J::IdealSheaf)
   return IdealSheaf(X, K, check=false)
 end
 
+
+@doc raw"""
+    IdealSheaf(X::NormalToricVariety, I::MPolyIdeal)
+
+Create a sheaf of ideals on a toric variety ``X`` from a homogeneous ideal 
+`I` in its `cox_ring`.
+
+# Examples
+```jldoctest
+julia> P3 = projective_space(NormalToricVariety, 3)
+Normal, non-affine, smooth, projective, gorenstein, fano, 3-dimensional toric variety without torusfactor
+
+julia> (x1,x2,x3,x4) = gens(cox_ring(P3))
+4-element Vector{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}:
+ x1
+ x2
+ x3
+ x4
+
+julia> I = ideal([x2,x3])
+ideal(x2, x3)
+
+julia> IdealSheaf(P3, I)
+Sheaf of ideals
+  on scheme over QQ covered with 4 patches
+    1: [x_1_1, x_2_1, x_3_1]   normal, affine toric variety
+    2: [x_1_2, x_2_2, x_3_2]   normal, affine toric variety
+    3: [x_1_3, x_2_3, x_3_3]   normal, affine toric variety
+    4: [x_1_4, x_2_4, x_3_4]   normal, affine toric variety
+with restrictions
+  1: ideal(x_2_1, x_3_1)
+  2: ideal(x_2_2, x_3_2)
+  3: ideal(1, x_3_3)
+  4: ideal(x_2_4, 1)
+```
+"""
+function IdealSheaf(X::NormalToricVariety, I::MPolyIdeal)
+  @req base_ring(I) === cox_ring(X) "ideal must live in the cox ring of the variety"
+
+  # We currently only support this provided that the following conditions are met:
+  # 1. All maximal cones are smooth, i.e. the fan is smooth/X is smooth.
+  # 2. The dimension of all maximal cones matches the dimension of the fan.
+  @req is_smooth(X) "Currently, ideal sheaves are only supported for smooth toric varieties"
+  @req all(m -> dim(m) == dim(X), maximal_cones(X)) "Currently, ideal sheaves require that all maximal cones have the dimension of the variety"
```

Comment:
```suggestion
  @req ispure(X) "Currently, ideal sheaves require that all maximal cones have the dimension of the variety"
```
That should create less intermediate objects.

Reply:
Thank you.